### PR TITLE
fix: don't link a device as its own via_device on a self-referencing batSn/parentSn

### LINKED ISSUE #731: HomeAssistantError: A device can not be its own via device during setup
### Describe the bug

Hello Veldkornet,

I am trying to set up my HYXI Halo HYX-MS3000AC via Modbus (RS485 USB). 
During the setup entry process, the integration crashes with a `HomeAssistantError`.

It seems like the integration is trying to assign a device as its own `via_device`, likely because the battery serial number (`bat_sn`) and the logger/inverter serial number (`sn`) match or overlap during registration in `_async_register_devices`.

Could you please take a look at the logic in `__init__.py` line 392 to prevent a device from referencing itself? 

Thank you for your amazing work on this integration!

**Error Log:**

### HYXi Device Type / Model

HYX-MS3000AC

### Home Assistant Version

2026.9.1

### Integration Version

v1.7.0 beta 6

### Debug Logs

```shell
Logger: homeassistant.config_entries
Quelle: config_entries.py:803
Erstmals aufgetreten: 20:53:36 (1 Vorkommnis)
Zuletzt protokolliert: 20:53:36

Error setting up entry HYXI Modbus (/dev/ttyUSB1) for hyxi_cloud
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 803, in __async_setup_with_context
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/hyxi_cloud/__init__.py", line 226, in async_setup_entry
    _async_register_devices(hass, entry, coordinator)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/hyxi_cloud/__init__.py", line 392, in _async_register_devices
    device_registry.async_get_or_create(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        config_entry_id=entry.entry_id,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        identifiers={(DOMAIN, bat_sn)},
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        via_device_id=devices[sn].id,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 2576, in async_get_or_create
    device = self._async_update_device(
        device.id,
    ...<9 lines>...
        **validated_fields,
    )
  File "/usr/src/homeassistant/homeassistant/helpers/device_registry.py", line 3097, in _async_update_device
    raise HomeAssistantError("A device can not be its own via device")
homeassistant.exceptions.HomeAssistantError: A device can not be its own via device
```

### PR DESCRIPTION
**Pull request by Veldkornet**
Fixes #731.

HYXI Halo all-in-one units (and potentially any device reporting a self-referencing `parentSn`) can report `batSn`/`parentSn` equal to their own `sn`. Registering that tried to set the device's `via_device_id` to its own device registry id, which Home Assistant's device registry rejects with `HomeAssistantError`, aborting config entry setup entirely -- matching the traceback in the issue.

## What changed

Guards against this everywhere a via-device link gets resolved: both passes of `_async_register_devices` (battery-to-inverter and child-to-parent-collector), the shared `via_device_id()` helper (now takes the caller's own `sn` and treats a self-match as unresolvable, consistent with its existing "not appropriate to link" contract), and `_battery_device_info`.

Also fixes a coordinator debug log that always claimed the polling interval was in minutes, even for Modbus entries where the same option means seconds -- unrelated to the crash, but caught while working the same file.

## Why it slipped past testing

The existing `_async_register_devices` unit tests mocked the device registry with `MagicMock()`, which accepts a self-referencing `via_device_id` without complaint -- only a real HA device registry rejects it. The real-`hass` integration suite had coverage for the EM-to-inverter and child-to-parent links, but none at all for the battery-to-inverter link, let alone this self-referencing edge case.

## Testing

Added real-registry integration tests for both the `batSn` and `parentSn` self-reference cases (they crash setup pre-fix), matching mock-based unit tests, and a direct test of `via_device_id`'s new guard.

Full CI-exact suite: `pytest tests/ --ignore=tests/integration` and `HYXI_INTEGRATION_TEST=1 pytest tests/integration`, both green, 100% coverage. Pre-commit clean.

## Summary by Sourcery

Prevent self-referencing device relationships from breaking setup while retaining valid device links and correcting polling-interval logging.

Bug Fixes:
- Prevent self-referencing `batSn` and `parentSn` values from creating invalid device-to-itself links and aborting Home Assistant config entry setup.
- Preserve normal battery and parent-collector device relationships while treating self-references as unlinked devices.

Enhancements:
- Correct coordinator polling-interval debug logs to report minutes for cloud entries and seconds for Modbus entries.

Tests:
- Add unit and real-registry integration coverage for distinct battery links and self-referencing battery and parent serials.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented devices from incorrectly linking to themselves as battery or parent devices.
  * Improved battery sensor associations when the battery and inverter share the same serial number.
  * Corrected device relationship handling for alarm and parent collector sensors.
  * Updated polling interval messages to display the correct time units for different connection types.

### CODE REVIEW

File: custom_components/hyxi_cloud/coordinator.py [hunk 1]
```diff
@@ -85,10 +86,15 @@ def __init__(self, hass: HomeAssistant, client: HyxiApiClient, entry: ConfigEntr
...
+        # The same "update_interval" option means minutes for the rate-limited
+        # cloud API but seconds for local Modbus polling (HyxiModbusCoordinator
+        # overrides self.update_interval accordingly right after this runs) --
+        # log it with the unit that will actually apply, not always "minutes".
         _LOGGER.debug(
-            "Initializing HYXI Coordinator for '%s' with polling interval: %s minutes",
+            "Initializing HYXI Coordinator for '%s' with polling interval: %s %s",
             entry.title,
             interval,
+            "seconds" if is_modbus_entry(entry) else "minutes",
```
[the remark this answers is not in the document]
Comment (Veldkornet) [reply custom_components/hyxi_cloud/coordinator.py]: Good catch -- fixed: the debug log now uses `entry.options.get("update_interval", DEFAULT_MODBUS_INTERVAL)` for Modbus entries, matching HyxiModbusCoordinator's own fallback, instead of the cloud default.
